src/inmanta/deploy/tasks.py

```diff
@@ -225,7 +249,10 @@ async def execute(self, task_manager: "scheduler.TaskManager", agent: str, reaso
         started = datetime.datetime.now().astimezone()
         try:
             my_executor: executor.Executor = await self.get_executor(
-                task_manager, agent, executor_resource_details.id.entity_type, self.version
+                task_manager=task_manager,
+                agent_spec=(agent, [executor_resource_details.id.entity_type]),
```

Comment:
any reason for not retrieving the entity type from `self.resource_intent` instead ?

Reply:
I think both are fine, I stuck with what was used on other lines.